SoliditybasicsCourse/contracts/OverrideCourse.sol

```diff
@@ -0,0 +1,28 @@
+// SPDX-License-Identifier: UNLICENSED
+
+pragma solidity 0.8.3;
+
+import {InterfaceContract} from "./interfaces/CourseInterfaces.sol";
+
+contract thisContract is InterfaceContract {
+    uint256[] valueArray;
+    mapping(uint256 => uint256) mappingAll;
+
+    function set(uint256 key, uint256 value) external override {
+        require(mappingAll[key] == 0);
+        mappingAll[key] = value;
+        valueArray.push(value);
+    }
```

Comment:
This function makes it possible to set a non-zero value only once. There's no way to change it afterwards.